README.md

```diff
@@ -32,13 +49,234 @@ import (
+https://demo.imgix.net/image.png?ar=4%3A3&dpr=2&h=800&q=50 2x
+https://demo.imgix.net/image.png?ar=4%3A3&dpr=3&h=800&q=35 3x
+https://demo.imgix.net/image.png?ar=4%3A3&dpr=4&h=800&q=23 4x
+https://demo.imgix.net/image.png?ar=4%3A3&dpr=5&h=800&q=20 5x
+```
+
+For more information to better understand srcset, we highly recommend
+[Eric Portis' "Srcset and sizes" article](https://ericportis.com/posts/2014/srcset-sizes/) which goes into depth about the subject.
+
+#### Variable Quality
+
+This library will automatically append a variable `q` parameter mapped to each `dpr` parameter when generating a [fixed-width image](#fixed-width-images) srcset. This technique is commonly used to compensate for the increased file size of high-DPR images.
+
+Since high-DPR images are displayed at a higher pixel density on devices, image quality can be lowered to reduce overall file size––without sacrificing perceived visual quality. For more information and examples of this technique in action, see [this blog post](https://blog.imgix.com/2016/03/30/dpr-quality).
+
+This behavior will respect any overriding `q` value passed in as a parameter. Additionally, it can be disabled altogether by passing `SrcSetOpts{disableVariableQuality: true}` to `CreateSrcSet`.
+
+
+### Fluid-Width Images
+
+#### Custom Widths
+
+In situations where specific widths are desired when generating `srcset` pairs, a user can specify them by passing an array of positive integers to `CreateSrcSetFromWidths`:
+
+```go
+ub := NewURLBuilder("demo.imgix.net")
+srcset := ub.CreateSrcSetFromWidths("image.jpg", url.Values{}, []int{100, 200, 300, 400})
+```
+```html
+https://demo.imgix.net/image.jpg?w=100 100w,
+https://demo.imgix.net/image.jpg?w=200 200w,
+https://demo.imgix.net/image.jpg?w=300 300w,
+https://demo.imgix.net/image.jpg?w=400 400w
+```
+
+#### Width Ranges
+
+In certain circumstances, you may want to limit the minimum or maximum value of the non-fixed `srcset` generated by the `CreateSrcSet` method. To do this, you can specify the widths at which a srcset should `begin` and `end` by specifying a `WidthRange` in your `SrcSetOpts`:
+
+```go
+ub := NewURLBuilder("demo.imgix.net")
+options := SrcSetOpts{widthRange: WidthRange{begin: 100, end: 380, tol: 0.08}}
+srcset := ub.CreateSrcSet("image.png", url.Values{}, options)
+```
+
+```html
+https://demo.imgix.net/image.png?w=100 100w,
+https://demo.imgix.net/image.png?w=116 116w,
+https://demo.imgix.net/image.png?w=135 135w,
+https://demo.imgix.net/image.png?w=156 156w,
+https://demo.imgix.net/image.png?w=181 181w,
+https://demo.imgix.net/image.png?w=210 210w,
+https://demo.imgix.net/image.png?w=244 244w,
+https://demo.imgix.net/image.png?w=283 283w,
+https://demo.imgix.net/image.png?w=328 328w,
+https://demo.imgix.net/image.png?w=380 380w
+```
+
+#### Width Tolerance
+
+The `srcset` width `tol`erance dictates the maximum `tol`erated difference between an image's downloaded size and its rendered size.
```

Comment:
Weak opinion here that the inline code-blocking of tol in `tol`erance leads to lower readability. My weak recommendation would be to only do it once, or instead use the abbrevation format commonly used in formal writing for the first time tolerance is mentioned: "The srcset width tolerance (`tol`) dictates the maximum tolerated difference ...".

Reply:
Yeah, I like that!